src/uct/ib/mlx5/ib_mlx5.h

```diff
@@ -618,7 +620,7 @@ uct_ib_mlx5_md_buf_free(uct_ib_mlx5_md_t *md, void *buf, uct_ib_mlx5_devx_umem_t
     }
 
     mlx5dv_devx_umem_dereg(mem->mem);
-    if (md->super.fork_init) {
+    if (md->super.fork_init && mem->size) {
```

Comment:
why needed?

Reply:
honestly md->super.fork_init won't always be same for alloc and for free

Reply:
how can it be? isn't it decided during md initialization?

Reply:
in uct_ib_md_open(), after uct_ib_md_ops->open() call. uct_ib_mlx5_md->zero_mem allocated in uct_ib_mlx5_devx_md_open() when fork_init not set yet